### PR TITLE
Check user related headers can be included independently.

### PR DESCRIPTION
**Pull request by ActoryOu**
Check user related headers can be included independently.

Description
-----------

As #959 mentioned, some header files need to be included in specific order.
This PR solves the header files that needs by user to make sure all headers can be included in any order.
And add a CI test flow to avoid this happening in the future.

Test Steps
-----------

Run new build-combination test.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/build-combination/CMakeLists.txt [hunk 1]
```diff
@@ -66,31 +75,38 @@ else()
     add_library( freertos_config ALIAS freertos_plus_tcp_config_default)
 endif()
 
-add_executable(freertos_plus_tcp_build_test EXCLUDE_FROM_ALL)
```
[the remark this answers is not in the document]
Comment (ActoryOu) [reply test/build-combination/CMakeLists.txt:69]: This one still exists in line 84. And this change adds a new one in headerSelfContain.cmake

[the remark this answers is not in the document]
Comment (moninom1) [reply test/build-combination/CMakeLists.txt:69]: Yeah it exists but seems like indentation is changed. We should not touch the code we are not changing

[the remark this answers is not in the document]
Comment (moninom1) [reply test/build-combination/CMakeLists.txt:69]: Got it. Seems fine.

**Review by tony-josi-aws — APPROVED** [review]

**Review by tony-josi-aws — DISMISSED** [review]

**Review by moninom1 — DISMISSED** [review]

File: test/build-combination/HeaderSelfContain/main.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (paulbartell) [inline test/build-combination/HeaderSelfContain/main.c:1]: Could this be handled in the cmake file instead? Seems like you could use the -include argument to gcc to achieve this.

File: test/build-combination/HeaderSelfContain/headerSelfContain.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
+set( COMPILE_OPTIONS
```
Comment (paulbartell) [inline test/build-combination/HeaderSelfContain/headerSelfContain.cmake:1]: target_compile_options could be used here.

**Review by shubnil — DISMISSED** [review]

File: test/build-combination/HeaderSelfContain/main.c (hunk 2 above)
Comment (ActoryOu) [reply test/build-combination/HeaderSelfContain/main.c]: That's a good idea to achieve this.
Updated PR by:
1. Add `-include ${header}.h` as compile option.
2. Remove redundant main.c and stubs.c and reuse main.c in Common directory.

Thanks Paul.

File: test/build-combination/HeaderSelfContain/headerSelfContain.cmake (hunk 3 above)
Comment (ActoryOu) [reply test/build-combination/HeaderSelfContain/headerSelfContain.cmake]: Updated

**Review by tony-josi-aws — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**